File: Task2/apz-pzpi-21-4-molchan-stanislav-task2/SmartWardrobeAnalytics/Domain/Entities/Item.cs
using Domain.Common;
using Domain.Enums;
using MongoDB.Bson;

namespace Domain.Entities;

public class Item : EntityBase
{
    public string Name { get; set; }
    
    public string Description { get; set; }
    
    public Categories Category { get; set; }
    
    public ObjectId BrandId { get; set; }
    
    public ObjectId CollectionId { get; set; }
}

File: Task4/apz-pzpi-21-4-molchan-stanislav-task4/SmartWardrobeAnalyticsWeb/src/app/models/item.ts
export interface Item {
    id: string
    name: string
    description: string
    categories: number
    brandId: string
    collectionId: string
}


export interface ItemCreate {
    name: string
    description: string
    categories: number
    brandId: string
    collectionId: string
}

export interface PaginatedResult<T> {
    items: T[];
    pageNumber: number;
    pageSize: number;
    totalPages: number;
    totalItems: number;
    hasPreviousPage: boolean;
    hasNextPage: boolean;
  }